### PR TITLE
Type checking and memory flags

### PR DESCRIPTION
**Pull request by amherag**
Fixes #

Changes:
- Added a better type checking mechanism
- Added command line flags for setting stack size, initial heap size and max heap size
- Removed the object-explorer/ directory. It's not going to be used
- Renamed README-images/ to readme-images/

Does this change need to mentioned in CHANGELOG.md?
Yes.

### CODE REVIEW

File: cx/makers.go [hunk 1]
```diff
@@ -77,23 +77,26 @@ func MakeArgument(name string, fileName string, fileLine int) *CXArgument {
...
 		offset += inp.Size
 		fn.Inputs = append(fn.Inputs, inp)
 	}
-	for _, typCode := range outputs {
-		fn.Outputs = append(fn.Outputs, MakeArgument("", "", -1).AddType(TypeNames[typCode]))
-		out := MakeArgument("", "", -1).AddType(TypeNames[typCode])
+	for _, out := range outputs {
+		// for _, typCode := range outputs {
+		// fn.Outputs = append(fn.Outputs, MakeArgument("", "", -1).AddType(TypeNames[typCode]))
+		// out := MakeArgument("", "", -1).AddType(TypeNames[typCode])
```
Comment (ingwal) [inline cx/makers.go]: Should these commented out lines still be there?

File: cx/op.go [hunk 2]
```diff
@@ -193,6 +193,7 @@ func MarkAndCompact() {
 // ResizeMemory ...
 func ResizeMemory(newMemSize int, isExpand bool) {
 	if newMemSize > MAX_HEAP_SIZE {
+		Debug("ENTEERIIING")
```
Comment (ingwal) [inline cx/op.go]: Slightly unhelpful debug message... :)

Comment (ingwal) [reply cx/op.go]: Bah!  Pressed the wrong button. This should just have been a comment, not a review.

File: cx/makers.go (hunk 1 above)
Comment (amherag) [reply cx/makers.go]: Nope. I think I'll have a comment-clean-up session soon. There are many code lines that I left "just in case".

**Pull request merged**